packages/core/src/interpreter.ts

```diff
@@ -880,7 +880,8 @@ export class Interpreter<
             ? serviceCreator(context, _event.data, {
                 data: resolvedData,
                 src: invokeSource,
-                meta: activity.meta
+                meta: activity.meta,
+                self: this
```

Comment:
Could we add `this.self = toActorRef(this)` to the constructor (the exact code might differ from this example) and pass that here?

We should limit what the user has access to from the callback - it's already limited on the type-level (since it got typed as `ActorRef<TEvent, State<TContext, TEvent>>`) but we should also limit it in the runtime.

Do we support restarting interpreters (as in `s.start(); s.stop(); s.start()`)? If yes then the `this.self` would have to be created in the `start` method.

We should probably also ensure that the `stop` method is not exposed on this (the `ActorRef` currently allows it). It sounds like a bad idea to allow a child to stop its parent imperatively, right?

Reply:
Done!